openupgradelib/openupgrade.py

```diff
@@ -2367,3 +2368,88 @@ def set_xml_ids_noupdate_value(env, module, xml_ids, value):
+
+def move_to_company_dependent(
+        env,
+        model_name,
+        model_table_name,
+        origin_field_name,
+        destination_field_name,
+    ):
+        """ For each row in a given table, the value of a given field is
+        set in another 'company dependant' field of the same table.
+        Useful in cases when from one version to another one, some field in a
+        model becomes a 'company dependent' field.
+
+        :param model_name: Name of the model.
+        :param model_table_name: Name of the table.
+        :param origin_field_name: Name of the field from which the values
+                                  will be obtained.
+        :param company_dependent_field_name: Name of the 'company dependent'
+                                             field where the values obtained
+                                             from origin_field_name will be
+                                             set.
+        """
+        logger.debug(
+            "Moving data to a company_dependent field:\n"
+            "   Model: {}\n"
+            "   Origin field name: {}\n"
+            "   Destination field name: {}".format(
+                model_name, origin_field_name, destination_field_name
+            )
+        )
+        cr = env.cr
+        mapping_type2field = {
+            'char': 'value_text',
+            'float': 'value_float',
+            'boolean': 'value_integer',
+            'integer': 'value_integer',
+            'text': 'value_text',
+            'binary': 'value_binary',
+            'many2one': 'value_reference',
+            'date': 'value_datetime',
+            'datetime': 'value_datetime',
+            'selection': 'value_text',
+        }
+        # Determine field id and the model name of the relation
+        # in case of many2one.
+        cr.execute("""
+            SELECT id, relation
+            FROM ir_model_fields
+            WHERE name='{}' AND model='{}'""".format(
+                destination_field_name, model_name))
+        destination_field_id, relation = cr.fetchone()
+        # Determine values. List of tuples that specifie what
+        # is the value for each row id in the table of model
+        # model_name.
+        env.cr.execute(
+            "SELECT id, {0} FROM {1} WHERE {0} IS NOT NULL".format(
+                origin_field_name, model_table_name))
+        values = env.cr.fetchall()
+        # Iterate values to insert into ir_property table
+        for model_id, value in values:
+            field_type = env[model_name]._fields[destination_field_name].type
+            # Determine extra field value: see mapping_type2field.
+            extra_field_value = value
+            extra_field = mapping_type2field.get(field_type)
+            if extra_field == 'value_reference':
+                extra_field_value = '%s,%d' % (relation, value)
+            # Insert into ir_property table using a sql query
+            logged_query(
```

Comment:
You must do everything in a SQL sentence. Imagine if you have 2 million records to find that 2 million lines of log.

You can perform `INSERT INTO ... SELECT ...` in the same SQL sentence for inserting all rows at the same time:

https://www.w3schools.com/sql/sql_insert_into_select.asp

but you have to deal with all the conditional in SQL. You have `CASE` for that.